src/components/VesselsListItem.vue

```diff
@@ -34,21 +35,27 @@
       }
     },
 
-    data () {
-      return {
-        inProgress: false
-      }
-    },
-
     computed: {
       vesselDetailsLink () {
         return `/dashboard/vessels/${this.vessel.id}`
+      },
+
+      inProgress () {
+        return this.deleteVesselTask.isActive
       }
     },
 
     methods: {
-      enableInProgress () {
-        this.inProgress = true
+      ...mapActions([
+        'deleteVessel'
+      ])
+    },
+
+    tasks (t) {
```

Comment:
Up there, where it appears this for the first time, we have `tasks (task) {....}` in contrast to `tasks (t) {` here. Maybe you want to keep it consistent, just a suggestion.